src/PowerShellEditorServices/Services/Analysis/PssaCmdletAnalysisEngine.cs

```diff
@@ -0,0 +1,544 @@
+            {
+                _logger.LogError("Formatter returned null result");
+                return null;
+            }
+
+            if (result.HasErrors)
+            {
+                var errorBuilder = new StringBuilder().Append(s_indentJoin);
+                foreach (ErrorRecord err in result.Errors)
+                {
+                    errorBuilder.Append(err).Append(s_indentJoin);
+                }
+                _logger.LogWarning($"Errors found while formatting file: {errorBuilder}");
+                return null;
+            }
+
+            foreach (PSObject resultObj in result.Output)
+            {
+                if (resultObj?.BaseObject is string formatResult)
+                {
+                    return formatResult;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Analyze a given script using PSScriptAnalyzer.
+        /// </summary>
+        /// <param name="scriptContent">The contents of the script to analyze.</param>
+        /// <returns>An array of markers indicating script analysis diagnostics.</returns>
+        public Task<ScriptFileMarker[]> AnalyzeScriptAsync(string scriptContent) => AnalyzeScriptAsync(scriptContent, settings: null);
+
+
+        /// <summary>
+        /// Analyze a given script using PSScriptAnalyzer.
+        /// </summary>
+        /// <param name="scriptContent">The contents of the script to analyze.</param>
+        /// <param name="settings">The settings file to use in this instance of analysis.</param>
+        /// <returns>An array of markers indicating script analysis diagnostics.</returns>
+        public Task<ScriptFileMarker[]> AnalyzeScriptAsync(string scriptContent, Hashtable settings)
+        {
+            // When a new, empty file is created there are by definition no issues.
+            // Furthermore, if you call Invoke-ScriptAnalyzer with an empty ScriptDefinition
+            // it will generate a ParameterBindingValidationException.
+            if (string.IsNullOrEmpty(scriptContent))
+            {
+                return Task.FromResult(Array.Empty<ScriptFileMarker>());
+            }
+
+            var command = new PSCommand()
+                .AddCommand("Invoke-ScriptAnalyzer")
+                .AddParameter("ScriptDefinition", scriptContent)
+                .AddParameter("Severity", s_scriptMarkerLevels);
+
+            object settingsValue = settings ?? _settingsParameter;
+            if (settingsValue != null)
+            {
+                command.AddParameter("Settings", settingsValue);
+            }
+            else
+            {
+                command.AddParameter("IncludeRule", _rulesToInclude);
+            }
+
+            return GetSemanticMarkersFromCommandAsync(command);
+        }
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Either implement 'IDisposable.Dispose', or totally rename this method to prevent confusion.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4961943)